src/FirebaseAuth.js

```diff
@@ -222,24 +222,23 @@
      */
     _routerMethodOnAuthPromise: function(rejectIfAuthDataIsNull) {
       var ref = this._ref;
-      var deferred = this._q.defer();
 
-      function callback(authData) {
-        if (authData !== null) {
-          deferred.resolve(authData);
-        } else if (rejectIfAuthDataIsNull) {
-          deferred.reject("AUTH_REQUIRED");
-        } else {
-          deferred.resolve(null);
+      return this._utils.promise(function(resolve,reject){
+        function callback(authData) {
+          if (authData !== null) {
+            resolve(authData);
```

Comment:
Chaining isn't at issue here (we aren't in `then` callback).

I made the changes requested. IntelliJ complains that the return statements are unnecessary, but the linter passes.